google/cloud/aiplatform/pipeline_jobs.py

```diff
@@ -240,6 +240,12 @@ def run(
 
         _LOGGER.log_create_with_lro(self.__class__)
 
+        if not sync:
```

Comment:
This block will not be executed because `sync` is currently not passed by the `base.optional_sync` decorator.

`sync` in the function signature for these methods is mainly provided as documentation for the caller and not currently intended to be used internally.

However, it should be easily supported by changing this line: 

https://github.com/googleapis/python-aiplatform/blob/ac181824ee5afd64640e1790e7ac70809c7cdb1f/google/cloud/aiplatform/base.py#L655

to 

```Python
sync = kwargs.get("sync", True)
```